odl/space/pspace.py

```diff
@@ -577,14 +578,28 @@ def __str__(self):
 
     def __repr__(self):
         """Return ``repr(self)``."""
+        # TODO: add the other optional arguments
         if self.size == 0:
-            return '{}(field={})'.format(self.__class__.__name__, self.field)
+            posargs = []
+            optargs = [('field', self.field, None)]
+            oneline = True
         elif self.is_power_space:
-            return '{}({!r}, {})'.format(self.__class__.__name__,
-                                         self.spaces[0], self.size)
+            posargs = [self.spaces[0], self.size]
+            optargs = []
+            oneline = True
         else:
-            inner_str = ', '.join(repr(space) for space in self.spaces)
+            posargs = self.spaces
+            optargs = []
+            argstr = ', '.join(repr(s) for s in self.spaces)
+            oneline = (len(argstr) <= 40 and '\n' not in argstr)
```

Comment:
I guess it is fine. Ideally we'd like a "global" optimization, but I guess that is not feasible so this is fine.